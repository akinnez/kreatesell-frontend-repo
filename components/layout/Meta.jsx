import Head from "next/head";

export const Meta = ({ title, keywords, description }) => {
  const name = "KreateSell";
  const websiteURL = "https://kreatesell.com";
  const twitterHandle = "@KreateSell";
  const metaContent = "";

  return (
    <Head>
      <meta data-charset="UTF-8" />
      <meta name="viewport" content="width=device-width, initial-scale=1.0" />
      <meta name="title" content={name} />
      <meta name="author" content={name} />
      <meta name="theme-color" content="#b8c1ec" />
      <meta name="pagename" content={metaContent} />
      <meta name="url" content={websiteURL} />
      <meta itemProp="name" content={name} />
      <meta property="og:site_name" content={name} />
      <meta property="og:type" content="website" />
      <meta property="og:url" content={websiteURL} />
      <meta property="og:title" content={name} />
      <meta property="og:description" content={metaContent} />
      <meta property="twitter:card" content="summary" />
      <meta property="twitter:url" content={websiteURL} />
      <meta property="twitter:title" content={name} />
      <meta name="twitter:site" content={twitterHandle} />
      <meta name="twitter:creator" content={twitterHandle} />
      {/* <meta property="twitter:description" content="Rocket Global" /> */}
      <meta property="twitter:image:alt" content={`A picture of ${name}`} />
      <meta name="keywords" content={keywords} />
      <meta name="description" content={description} />
      <link rel="icon" href="/favicon.ico" />
      <title>{title}</title>
    </Head>
  );
};

Meta.defaultProps = {
  title: "KreateSell",
  keywords: "e-commerce",
  description: "KreateSell , Sell more, Do less",
};
