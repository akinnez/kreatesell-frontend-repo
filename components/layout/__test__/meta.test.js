import renderer from "react-test-renderer";
import { Meta } from "../Meta";

it("renders a snapshot of <Meta />", () => {
  const metaElement = renderer
    .create(
      <Meta
        title="mockTitle"
        description="This is  a mock description"
        keywords="mockKeyword"
      />
    )
    .toJSON();
  expect(metaElement).toMatchSnapshot();
});
