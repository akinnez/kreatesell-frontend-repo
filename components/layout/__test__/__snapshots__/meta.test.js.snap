// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot of <Meta /> 1`] = `
Array [
  <meta
    data-charset="UTF-8"
  />,
  <meta
    content="width=device-width, initial-scale=1.0"
    name="viewport"
  />,
  <meta
    content="KreateSell"
    name="title"
  />,
  <meta
    content="KreateSell"
    name="author"
  />,
  <meta
    content="#b8c1ec"
    name="theme-color"
  />,
  <meta
    content=""
    name="pagename"
  />,
  <meta
    content="https://kreatesell.com"
    name="url"
  />,
  <meta
    content="KreateSell"
    itemProp="name"
  />,
  <meta
    content="KreateSell"
    property="og:site_name"
  />,
  <meta
    content="website"
    property="og:type"
  />,
  <meta
    content="https://kreatesell.com"
    property="og:url"
  />,
  <meta
    content="KreateSell"
    property="og:title"
  />,
  <meta
    content=""
    property="og:description"
  />,
  <meta
    content="summary"
    property="twitter:card"
  />,
  <meta
    content="https://kreatesell.com"
    property="twitter:url"
  />,
  <meta
    content="KreateSell"
    property="twitter:title"
  />,
  <meta
    content="@KreateSell"
    name="twitter:site"
  />,
  <meta
    content="@KreateSell"
    name="twitter:creator"
  />,
  <meta
    content="Rocket Global"
    property="twitter:description"
  />,
  <meta
    content="A picture of KreateSell"
    property="twitter:image:alt"
  />,
  <meta
    content="mockKeyword"
    name="keywords"
  />,
  <meta
    content="This is  a mock description"
    name="description"
  />,
  <link
    href="/favicon.ico"
    rel="icon"
  />,
  <title>
    mockTitle
  </title>,
]
`;
